### PR TITLE
test that all configured buckets serve a sample blob

### PR DESCRIPTION
**Pull request by BenTheElder**
should help prevent adopting unconfigured buckets

### CODE REVIEW

**Comment by BenTheElder** [comment]
> in
cmd/archeio/internal/app/buckets_integration_test.go:120:14: G107: Potential HTTP request made with variable url (gosec)
>			r, err := http.Get(url)

Starting to think `gosec` is just a touch overzealous 🙄

File: cmd/archeio/internal/app/buckets_integration_test.go [hunk 1]
```diff
@@ -35,18 +43,18 @@ func TestCachedBlobChecker(t *testing.T) {
...
 		{
 			Name:         "same-known bucket entry",
-			BlobURL:      bucket + "/containers/images/sha256%3Ada86e6ba6ca197bf6bc5e9d900febd906b133eaa4750e6bed647b0fbe50ed43e",
+			BlobURL:      bucket + "/containers/images/sha256:da86e6ba6ca197bf6bc5e9d900febd906b133eaa4750e6bed647b0fbe50ed43e",
 			ExpectExists: true,
 		},
 		{
 			Name:         "known bucket, bad entry",
-			BlobURL:      bucket + "/c0ntainers/images/sha256%3Ada86e6ba6ca197bf6bc5e9d900febd906b133eaa4750e6bed647b0fbe50ed43e",
+			BlobURL:      bucket + "/c0ntainers/images/sha256:da86e6ba6ca197bf6bc5e9d900febd906b133eaa4750e6bed647b0fbe50ed43e",
```
Comment (BenTheElder) [inline cmd/archeio/internal/app/buckets_integration_test.go:57]: consistency with other test data / production code, on the off chance a backend actually handles this differently (it shouldn't ...)

File: cmd/archeio/internal/app/buckets_integration_test.go [hunk 2]
```diff
@@ -20,10 +20,18 @@ limitations under the License.
...
+	"encoding/hex"
+	"io"
+	"net/http"
 	"testing"
+
+	"k8s.io/registry.k8s.io/pkg/net/cloudcidrs"
 )
 
-func TestCachedBlobChecker(t *testing.T) {
+func TestIntegrationCachedBlobChecker(t *testing.T) {
```
Comment (BenTheElder) [inline cmd/archeio/internal/app/buckets_integration_test.go:33]: integration tests should be prefixed `TestIntegration` for running `make integration`

File: cmd/archeio/internal/app/buckets_integration_test.go (hunk 1 above)
Comment (dims) [reply cmd/archeio/internal/app/buckets_integration_test.go:57]: 👍🏾

**Pull request merged**